src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -951,6 +958,29 @@ internal void ResolveConflict([NotNull]string docID)
                 }
             });
         }
+
+        internal void SchedulePurgeExpired(TimeSpan delay)
+        {
+            var nextExpiration = Native.c4db_nextDocExpiration(_c4db);
+            if (nextExpiration > 0) {
+                var delta = (DateTimeOffset.FromUnixTimeMilliseconds((long)nextExpiration) - DateTimeOffset.UtcNow).Add(TimeSpan.FromSeconds(1));
+                var expirationTimeSpan = delta > delay ? delta : delay;
+                if (expirationTimeSpan.TotalMilliseconds >= UInt32.MaxValue) {
+                    _expirePurgeTimer.Change(TimeSpan.FromMilliseconds(UInt32.MaxValue - 1), TimeSpan.FromMilliseconds(-1));
+                    Log.To.Database.I(Tag, "{0:F3} seconds is too far in the future to schedule a document expiration," +
+                                           " will run again at the maximum value of {0:F3} seconds", expirationTimeSpan.TotalSeconds, (UInt32.MaxValue - 1) / 1000);
+                } else if (expirationTimeSpan.TotalSeconds <= Double.Epsilon) {
+                    _expirePurgeTimer.Change(Timeout.Infinite, Timeout.Infinite);
+                    PurgeExpired(null);
+                } else {
+                    _expirePurgeTimer.Change(expirationTimeSpan, TimeSpan.FromMilliseconds(-1));
```

Comment:
You should probably check what the existing timer is set to as well, to avoid needless calculation.  If the timer is already going to run sooner than the new delay then there is no need to reschedule.

Reply:
how to get current expiration timer's total milliseconds from _expirePurgeTimer?

Reply:
I thought there was a way but I was wrong, so this thread is not irrelevant.  Just double check on the `else if` condition here. Should it be milliseconds instead of seconds?